other/word_search.py

```diff
@@ -0,0 +1,395 @@
+        """
+        >>> ws = WordSearch(WORDS, 3, 3)
+        >>> ws.insert_north_east("cat", [2], [0])
+        >>> ws.board  # doctest: +NORMALIZE_WHITESPACE
+        [[None, None, 't'],
+        [None, 'a', None],
+        ['c', None, None]]
+        >>> ws.insert_north_east("at", [0, 1], [2, 1, 0])
+        >>> ws.board  # doctest: +NORMALIZE_WHITESPACE
+        [[None, 't', 't'],
+        ['a', 'a', None],
+        ['c', None, None]]
+        """
+        word_length = len(word)
+        # Attempt to insert the word into each row and when successful, exit
+        for row in rows:
+            # Check if there is space for the word above the row
+            if word_length > row + 1:
+                continue
+
+            # Attempt to insert the word into each column
+            for col in cols:
+                # Check if there is space to the right of the word as well as above
+                if word_length + col > self.width:
+                    continue
+
+                # Check if there are existing letters
+                # to the right of the column that will be overwritten
+                letters_diagonal_left = [
+                    self.board[row - i][col + i] for i in range(word_length)
+                ]
+                if all(letter is None for letter in letters_diagonal_left):
+                    # Successful, insert the word north
+                    for i in range(word_length):
+                        self.board[row - i][col + i] = word[i]
+                    return
+
+    def insert_east(self, word: str, rows: list[int], cols: list[int]) -> None:
+        """
+        >>> ws = WordSearch(WORDS, 3, 3)
+        >>> ws.insert_east("cat", [1], [0])
+        >>> ws.board  # doctest: +NORMALIZE_WHITESPACE
+        [[None, None, None],
+        ['c', 'a', 't'],
+        [None, None, None]]
+        >>> ws.insert_east("at", [1, 0], [2, 1, 0])
+        >>> ws.board  # doctest: +NORMALIZE_WHITESPACE
+        [[None, 'a', 't'],
+        ['c', 'a', 't'],
+        [None, None, None]]
+        """
+        word_length = len(word)
+        # Attempt to insert the word into each row and when successful, exit
+        for row in rows:
+            # Attempt to insert the word into each column
+            for col in cols:
+                # Check if there is space to the right of the word
+                if word_length + col > self.width:
+                    continue
+
+                # Check if there are existing letters
+                # to the right of the column that will be overwritten
+                letters_left = [self.board[row][col + i] for i in range(word_length)]
+                if all(letter is None for letter in letters_left):
+                    # Successful, insert the word north
+                    for i in range(word_length):
+                        self.board[row][col + i] = word[i]
+                    return
+
+    def insert_south_east(self, word: str, rows: list[int], cols: list[int]) -> None:
```

Comment:
```suggestion
    def insert_southeast(self, word: str, rows: list[int], cols: list[int]) -> None:
```
Nitpick: `south_east` -> `southeast`